requests/utils.py

```diff
@@ -679,28 +679,28 @@ def should_bypass_proxies(url, no_proxy):
     no_proxy_arg = no_proxy
     if no_proxy is None:
         no_proxy = get_proxy('no_proxy')
-    netloc = urlparse(url).netloc
+    parsed = urlparse(url)
 
     if no_proxy:
         # We need to check whether we match here. We need to see if we match
-        # the end of the netloc, both with and without the port.
+        # the end of the hostname, both with and without the port.
         no_proxy = (
             host for host in no_proxy.replace(' ', '').split(',') if host
         )
 
-        ip = netloc.split(':')[0]
-        if is_ipv4_address(ip):
+        if is_ipv4_address(parsed.hostname):
             for proxy_ip in no_proxy:
                 if is_valid_cidr(proxy_ip):
-                    if address_in_network(ip, proxy_ip):
+                    if address_in_network(parsed.hostname, proxy_ip):
                         return True
-                elif ip == proxy_ip:
+                elif parsed.hostname == proxy_ip:
                     # If no_proxy ip was defined in plain IP notation instead of cidr notation &
                     # matches the IP of the index
                     return True
         else:
+            host_with_port = '{}:{}'.format(parsed.hostname, parsed.port)
```

Comment:
A couple notes here. The `port` attribute is either an `int` or `None`, which means a case like http://localhost being passed to `urlparse` would yield `localhost:None` here. While that's unlikely to affect matching, I'm not sure that's what we want to be doing.

The second thing is we're still planning to support python 2.6 until 3.0.0 is released which means this format statement won't work as is. We'll need to provide something along the lines of `{0}:{1}` instead to avoid the ValueError.